Add unit testing

nnist opened the pull request:
This adds unit tests using the standard `unittest` library. It currently has 89% code coverage. By adding this to your project, future development, refactoring, etc. is much less likely to break existing functionality.

I couldn't make tests for the vertical graph functionality because it doesn't work for me. It gives `NameError: name 'args' is not defined` in `vertically`.

You can run the tests with `python3 -m unittest` in the root directory of the project. The test coverage report can be generated with the helper script I added.

mkaz commented:
This looks good, thanks for rebasing, and adding tests. Its nice to know when making changes, we're not breaking things.

I couldn't get the test coverage script to work, I installed `pip3 install coverage` but still says cannot find `python3-coverage`

Also, when I run the tests, one test still fails the rest all pass

```
FAIL: test_calendar_heatmap_without_start_date_prints_correctly (tests.test_termgraph.TermgraphTest)
```

nnist commented:
I fixed the failing test `test_calendar_heatmap_without_start_date_prints_correctly`, I think I forgot to change the expected output when some previous pull requests were merged.
Also fixed the tests for #35 / #36.

It's strange that coverage doesn't work for you, what happens when you just try to run `coverage` instead?
